trailmet/algorithms/prune/network_slimming.py

```diff
@@ -96,14 +73,13 @@ def __init__(self,**kwargs):
         self.epochs = kwargs.get('epochs',200)
         self.s = kwargs.get('s' , 1e-3)
         self.lr = kwargs.get('learning_rate',2e-3)
-        #self.save_dir = kwargs.get('save_dir')
         self.prune_schema = os.path.join(kwargs.get('schema_root') , f"schema/{self.net}.json")
         self.sparsity_train = kwargs.get('sparsity_train',True)
         self.fine_tune_epochs = kwargs.get('fine_tune_epochs',165)
         self.fine_tune_lr = kwargs.get('fine_tune_learning_rate',1e-4)
         self.prune_ratio = kwargs.get('prune_ratio',0.5)
-#        self.bn_weight_vis_period = kwargs.get('bn_weight_vis_period')
         self.log_base = f"{self.net}_{self.dataset}.pth"
+        self.model = Model(self.net,self.num_classes)
```

Comment:
Why are we using a custom Model class here, cant we use the generic one ?

Reply:
I also agree that the custom model class can be removed, and a generic one can be used. I have used the generic model class for the torch pruning code. Let me know if the same fix should be made in the network slimming original code.